### PR TITLE
Add image classification example

### PR DESCRIPTION
**Pull request by laurentgrangeau**
This PR add an example with Jupyter notebook for image classification deployed on top of the blueprint

### CODE REVIEW

File: .github/dependabot.yaml [hunk 1]
```diff
@@ -34,11 +34,11 @@ updates:
     package-ecosystem: "pip"
     schedule:
       interval: "daily"
-  - directory: "/examples/federated-learning/tff/image-classification"
+  - directory: "/examples/federated-learning/tff/federated-learning-for-image-classification"
```
Comment (ferrarimarco) [inline .github/dependabot.yaml:37]: Add `/container-image` to this path

File: .github/dependabot.yaml [hunk 2]
```diff
@@ -34,11 +34,11 @@ updates:
...
     package-ecosystem: "docker"
     schedule:
       interval: "daily"
-  - directory: "/examples/federated-learning/tff/image-classification"
+  - directory: "/examples/federated-learning/tff/federated-learning-for-image-classification"
```
Comment (ferrarimarco) [inline .github/dependabot.yaml:41]: Add `/container-image` to this path

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,10 @@
+FROM quay.io/jupyter/tensorflow-notebook:tensorflow-2.15.0
+
+WORKDIR "/home/jovyan"
```
Comment (ferrarimarco) [inline examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:3]: Can you add a comment explaining why we set this value?

File: examples/federated-learning/tff/federated-learning-for-image-classification/workload-pkg/deployment.yaml [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+  template:
+    metadata:
+      labels:
+        app: fl-image-classification
+    spec:
+      containers:
+        - name: fl-image-classification
+          image: fl-image-classification:latest  # kpt-set: ${tff-runtime-container-image-id}
+          ports:
+            - containerPort: 8888
```
Comment (ferrarimarco) [inline examples/federated-learning/tff/federated-learning-for-image-classification/workload-pkg/deployment.yaml:37]: Do we need a corresponding `Service` for this `Deployment`?

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+
+COPY requirements.txt .
+RUN pip install --no-cache-dir -r requirements.txt
```
Comment (ferrarimarco) [inline examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:6]: Refactor to:

```shell
RUN pip install \
    --no-cache-dir \
    --requirement requirements.txt \
    && rm requirements.txt
```

for consistency with other container images.

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile [hunk 6]
```diff
@@ -0,0 +1,10 @@
...
+
+COPY . .
+
```
Comment (ferrarimarco) [inline examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:9]: Maybe we should define an `ENTRYPOINT` and a default `CMD`?

**Review by ferrarimarco — CHANGES_REQUESTED** [review]
Please, fix linting errors as well.

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile (hunk 6 above)
Comment (laurentgrangeau) [reply examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:9]: `ENTRYPOINT` and `CMD` are already defined in the base image

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile (hunk 3 above)
Comment (laurentgrangeau) [reply examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:3]: Maybe we don't need it at all as it is set in the base image

https://github.com/jupyter/docker-stacks/blob/main/images/docker-stacks-foundation/Dockerfile#L144

File: .github/dependabot.yaml (hunk 1 above)
Comment (laurentgrangeau) [reply .github/dependabot.yaml]: Done

File: .github/dependabot.yaml (hunk 2 above)
Comment (laurentgrangeau) [reply .github/dependabot.yaml]: Done

File: examples/federated-learning/tff/federated-learning-for-image-classification/workload-pkg/deployment.yaml (hunk 4 above)
Comment (laurentgrangeau) [reply examples/federated-learning/tff/federated-learning-for-image-classification/workload-pkg/deployment.yaml:37]: Done

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile (hunk 5 above)
Comment (laurentgrangeau) [reply examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile]: Done

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile (hunk 3 above)
Comment (ferrarimarco) [reply examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:4]: If it's not needed, remove.

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile (hunk 5 above)
Comment (ferrarimarco) [reply examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile]: Please, use this form with long options and line breaks so it's more readable.

**Review by ferrarimarco — CHANGES_REQUESTED** [review]

File: examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile (hunk 3 above)
Comment (laurentgrangeau) [reply examples/federated-learning/tff/federated-learning-for-image-classification/container-image/Dockerfile:4]: It's needed as per Superlinter. Will add a comment on what it is

File: .editorconfig [hunk 7]
```diff
@@ -30,7 +30,7 @@ indent_size = 2
 indent_size = unset
 
 # Don't try to format binary files
-[*.tfvars.enc]
+[*.{tfvars.enc,ipynb}]
```
Comment (ferrarimarco) [inline .editorconfig:33]: Do you really need to unset everything, or just indentation maybe?

This particular case is for binary files.

**Review by ferrarimarco — APPROVED** [review]

**Pull request closed without merging**